test/e2e/dryrun.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"sigs.k8s.io/kwok/pkg/utils/path"
+)
+
+func loadExpectedClusterDetails(filepath string) (string, error) {
+	data, err := os.ReadFile(filepath)
+	if err != nil {
+		return "", err
+	}
+	out := string(data)
+	out = strings.ReplaceAll(out, "\n", " ")
+	out = strings.Join(strings.Fields(out), " ")
+	return out, nil
+}
+
+func formatCmdOutput(output, clusterName, rootDir string) string {
+	got := output
+	extensions := map[string]string{
+		"windows": "zip",
+		"linux":   "tar.gz",
+		"darwin":  "tar.gz",
+	}
+	got = strings.ReplaceAll(got, clusterName, "<CLUSTER_NAME>")
+	got = strings.ReplaceAll(got, rootDir, "<ROOT_DIR>")
+	got = strings.ReplaceAll(got, runtime.GOOS, "<OS>")
+	got = strings.ReplaceAll(got, runtime.GOARCH, "<ARCH>")
+	got = strings.ReplaceAll(got, extensions[runtime.GOOS], "<TAR>")
+	got = strings.ReplaceAll(got, "\n", " ")
+	got = strings.Join(strings.Fields(got), " ")
+	return got
+}
+
+func CaseDryrun(clusterName string, kwokctlPath string, rootDir string, clusterRuntime string, updateTestdata bool) *features.FeatureBuilder {
```

Comment:
The logic of these 3 cases is almost the same, only the args and output path are different, you can try to make it into a function, for the 3 cases to call?

Reply:
done